### DISCUSSION TITLE
How to link to another javascript file

### DISCUSSION THREAD

**Discussion by KoolKreator** [community/community#23942] [Other Feature Feedback, Questions, & Ideas]
<p>Hello there I’m new to using Github pages and I’m having trouble getting an important dependency to work. I have the dependency source code and I have tried linking to that file as I did in my ide which worked there but not on Github is there a certain path that Github uses or am I doing something wrong?</p>
<p>Code I used:</p>
<pre><code class="lang-auto">import AudioMotionAnalyzer from "./src/audioMotion-analyzer"
</code></pre>
<h1>Git Repo:</h1>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/KoolKreator/AudiomotionMusicMixer" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/387;"><img src="https://user-images.githubusercontent.com/79553405/181080335-101c48a5-b430-45b5-8117-91493f519ce0.png" class="thumbnail" width="690" height="387" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/9/b91ca5669207d785c80d529dff49635132f38854_2_690x387.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/9/b91ca5669207d785c80d529dff49635132f38854_2_1035x580.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/b/9/b91ca5669207d785c80d529dff49635132f38854.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/9/b91ca5669207d785c80d529dff49635132f38854_2_10x10.png"></div>

<h3><a href="https://github.com/KoolKreator/AudiomotionMusicMixer" target="_blank" rel="noopener nofollow ugc">KoolKreator/AudiomotionMusicMixer</a></h3>

  <p><span class="label1">Contribute to KoolKreator/AudiomotionMusicMixer development by creating an account on GitHub.</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<h1>Github Pages Site:</h1>
<p><a href="https://koolkreator.github.io/AudiomotionMusicMixer/" class="onebox" target="_blank" rel="noopener nofollow ugc">https://koolkreator.github.io/AudiomotionMusicMixer/</a></p>

**Comment by otter-computer** [comment, marked as answer]
<p>Hmm it looks like you’re using npm to manage these dependencies for this project, is that right? GitHub Pages doesn’t natively support npm so it won’t install these for you when deploying your site.</p>
<p>When deploying a site like this you’ll usually have to install these and run your build script locally and then push the built version up to a separate branch on GitHub–usually the <code>gh-pages</code> branch. There’s a <a href="https://www.npmjs.com/package/gh-pages">useful module called <code>gh-pages</code></a> that can help with this deployment part. Once you’ve done that, and pushed your code to the <code>gh-pages</code> branch, you’ll need to set that new branch to the publishing source of your site in the repository settings.</p>
<p>You’ll also need to remember to run this build &amp; deploy process each time you make a change to your code!</p>

**Comment by KoolKreator** [comment]
<p>Thank you for your help I will try that out, also if I have the source code available would it be possible to use the dependency that way as well</p>
